gateway/gateway.go

```diff
@@ -586,14 +583,42 @@ func (gateway *HandleT) beaconBatchHandler(w http.ResponseWriter, r *http.Reques
 	gateway.beaconHandler(w, r, "batch")
 }
 
+func (gateway *HandleT) checkAndValidateWebHandler(w http.ResponseWriter, r *http.Request, reqType string) string {
+	var sourceFailStats = make(map[string]int)
+	var errorMessage = ""
+	writeKey, _, ok := r.BasicAuth()
+	if !ok || writeKey == "" {
+		errorMessage = response.GetStatus(response.NoWriteKeyInBasicAuth)
+		misc.IncrementMapByKey(sourceFailStats, "noWriteKey", 1)
+		gateway.updateSourceStats(sourceFailStats, "gateway.write_key_failed_requests")
+		return errorMessage
+	}
+	payload, err := gateway.getPayloadFromRequest(r)
+	if err == nil {
+		if reqType == "import" {
```

Comment:
As mentioned in my other comment, this check seems out of place here. I think we should directly handle it by sending the request to its own handler instead of webhandler